imageslider/src/main/ets/components/MainPage/MainPage.ets

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2022 Application Library Engineering Group.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ImageSliderModel } from './ImageSliderModel'
+
+@Component
+export struct ImageSlider {
+  @State obj: ImageSliderModel = new ImageSliderModel();
+  @State inSetValue: number = 0;
+  img: ResourceStr[];
+
+  aboutToAppear() {
+    if (this.img == null || this.img == undefined) {
+      console.error("please pass image resource array")
+    }
+  }
+
+  build() {
+    Column({ space: 30 }) {
```

Comment:
handle obj null & undefined validation
if obj is null don't build any ui else set default model in obj and show the default design